test/board/devices/input/switches.test.ts

```diff
@@ -0,0 +1,256 @@
+    expect(switches.isOn(14)).toBe(true)
+    expect(switches.isOn(15)).toBe(false)
+
+    expect(switches.isOn(16)).toBe(true)
+    expect(switches.isOn(17)).toBe(false)
+    expect(switches.isOn(18)).toBe(true)
+    expect(switches.isOn(19)).toBe(false)
+    expect(switches.isOn(20)).toBe(true)
+    expect(switches.isOn(21)).toBe(false)
+    expect(switches.isOn(22)).toBe(true)
+    expect(switches.isOn(23)).toBe(false)
+
+    expect(switches.isOn(24)).toBe(true)
+    expect(switches.isOn(25)).toBe(false)
+    expect(switches.isOn(26)).toBe(true)
+    expect(switches.isOn(27)).toBe(false)
+    expect(switches.isOn(28)).toBe(true)
+    expect(switches.isOn(29)).toBe(false)
+    expect(switches.isOn(30)).toBe(true)
+    expect(switches.isOn(31)).toBe(false)
+  })
+})
+
+describe('test set() function', () => {
+  test('test passing invalid positions to set() function', () => {
+    expect(() => switches.set(32, true)).toThrow(
+      new Error('Position 32 does not exist.')
+    )
+    expect(() => switches.set(-2, false)).toThrow(
+      new Error('Position -2 does not exist.')
+    )
+  })
+  test('test passing valid positions to set() function', () => {
+    switches.set(1, true)
+    switches.set(3, true)
+    switches.set(5, true)
+    switches.set(7, true)
+
+    switches.set(8, false)
+    switches.set(10, false)
+    switches.set(12, false)
+    switches.set(14, false)
+
+    switches.set(17, true)
+    switches.set(19, true)
+    switches.set(21, true)
+    switches.set(23, true)
+
+    switches.set(24, false)
+    switches.set(26, false)
+    switches.set(28, false)
+    switches.set(30, false)
+
+    expect(switches.readByte(block1)).toEqual(byte_1111_1111)
+    expect(switches.readByte(block2)).toEqual(byte_0000_0000)
+    expect(switches.readByte(block3)).toEqual(byte_1111_1111)
+    expect(switches.readByte(block4)).toEqual(byte_0000_0000)
+
+    expect(switches.isOn(0)).toBe(true)
+    expect(switches.isOn(1)).toBe(true)
+    expect(switches.isOn(2)).toBe(true)
+    expect(switches.isOn(3)).toBe(true)
+    expect(switches.isOn(4)).toBe(true)
+    expect(switches.isOn(5)).toBe(true)
+    expect(switches.isOn(6)).toBe(true)
+    expect(switches.isOn(7)).toBe(true)
+
+    expect(switches.isOn(8)).toBe(false)
+    expect(switches.isOn(9)).toBe(false)
+    expect(switches.isOn(10)).toBe(false)
+    expect(switches.isOn(11)).toBe(false)
+    expect(switches.isOn(12)).toBe(false)
+    expect(switches.isOn(13)).toBe(false)
+    expect(switches.isOn(14)).toBe(false)
+    expect(switches.isOn(15)).toBe(false)
+
+    expect(switches.isOn(16)).toBe(true)
+    expect(switches.isOn(17)).toBe(true)
+    expect(switches.isOn(18)).toBe(true)
+    expect(switches.isOn(19)).toBe(true)
+    expect(switches.isOn(20)).toBe(true)
+    expect(switches.isOn(21)).toBe(true)
+    expect(switches.isOn(22)).toBe(true)
+    expect(switches.isOn(23)).toBe(true)
+
+    expect(switches.isOn(24)).toBe(false)
+    expect(switches.isOn(25)).toBe(false)
```

Comment:
is this not already covered with the test `'test passing valid positions to isOn() function'` ?

Reply:
Yep, but I think this is a simple way to visualize the current state of the switches in the test. IMO we can leave it like this.

Reply:
Yes I wrote this test, my motto was "the more the better" does not hurt anyone. :D